### PR TITLE
common test subrange, take, drop

### PR DESCRIPTION
**Pull request by rscohn2**
take is broken in MHP and disabled

### CODE REVIEW

File: test/gtest/include/common/view-tests.hpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  DV dv_a(n);
+  TypeParam::iota(dv_a, 100);
+  TypeParam::for_each(TypeParam::policy(),
+                      rng::subrange(dv_a.begin() + 1, dv_a.end() - 1), neg);
+
+  if (comm_rank == 0) {
+    V a(n), a_in(n);
+    rng::iota(a, 100);
+    rng::iota(a_in, 100);
+    rng::for_each(rng::subrange(a.begin() + 1, a.end() - 1), neg);
```
Comment (lslusarczyk) [inline test/gtest/include/common/view-tests.hpp:21]: We do not test `for_each`. Let's remove its usage from `subrange` tests. See [unittests of subrange](https://github.com/ericniebler/range-v3/blob/master/test/view/subrange.cpp) for examples how to write simple checks on subrange results (check begin/end, dereference).

File: test/gtest/include/common/view-tests.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    EXPECT_TRUE(unary_check(a_in, a, dv_a));
```
Comment (lslusarczyk) [inline test/gtest/include/common/view-tests.hpp:22]: Check look perfectly readable when having exact values which we expect, like
```
EXPECT_TRUE(equal(dist_vec, V{2, 3, 4}));
```
... so one has no to trace what happened to to a_in, a, dv_a during the test and what does `unary_check` do.
See [slice unittest](https://github.com/ericniebler/range-v3/blob/master/test/view/slice.cpp) for a good example on how checks on expected results can look like.

File: test/gtest/include/common/view-tests.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  auto neg = [](auto &v) { v = -v; };
+  DV dv_a(n);
+  TypeParam::iota(dv_a, 100);
+  TypeParam::for_each(TypeParam::policy(), rng::views::drop(dv_a, 2), neg);
+
+  if (comm_rank == 0) {
+    V a(n), a_in(n);
+    rng::iota(a, 100);
+    rng::iota(a_in, 100);
+    rng::for_each(rng::views::drop(a, 2), neg);
```
Comment (lslusarczyk) [inline test/gtest/include/common/view-tests.hpp:41]: Same comments as for `subrange`. Ideally there should be no `take`, `iota`, `foreach` here, just `drop` and simple begin/end/dereference/equality checks

File: test/gtest/include/common/view-tests.hpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    EXPECT_TRUE(unary_check(a_in, a, dv_a));
+  }
+}
+
+TYPED_TEST_P(CommonTests, DISABLED_Take) {
```
Comment (lslusarczyk) [inline test/gtest/include/common/view-tests.hpp:46]: Why is this test commented out? Could we establish a rule, that we comment out a test always with a comment saying about the reason?

File: test/gtest/include/common-tests.hpp [hunk 5]
```diff
@@ -101,5 +101,7 @@ TYPED_TEST_SUITE_P(CommonTests);
 
 #include "common/algorithm-tests.hpp"
 #include "common/dv-tests.hpp"
-REGISTER_TYPED_TEST_SUITE_P(CommonTests, DistributedVectorRequirements,
-                            DistributedVectorConstructors, ForEach);
+#include "common/view-tests.hpp"
+REGISTER_TYPED_TEST_SUITE_P(CommonTests, DistributedVectorConstructors,
```
[the remark this answers is not in the document]
Comment (rscohn2) [reply test/gtest/include/common-tests.hpp:105]: ok

File: test/gtest/include/common/view-tests.hpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+TYPED_TEST_P(CommonTests, Subrange) {
+  using DV = typename TypeParam::DV;
```
[the remark this answers is not in the document]
Comment (rscohn2) [reply test/gtest/include/common/view-tests.hpp:6]: I agree in general. In this case, it is the `typename` that was the motivation for separate definition.

File: test/gtest/include/common/view-tests.hpp [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  using V = typename TypeParam::V;
+
+  std::size_t n = 10;
+
+  auto neg = [](auto &v) { v = -v; };
+  DV dv_a(n);
+  TypeParam::iota(dv_a, 100);
```
[the remark this answers is not in the document]
Comment (rscohn2) [reply test/gtest/include/common/view-tests.hpp:13]: Using initializer lists everywhere would be inflexible and tedious. We need a standard way to initialize. I don't like fill because it does not catch position errors.

File: test/gtest/include/common/view-tests.hpp (hunk 2 above)
Comment (rscohn2) [reply test/gtest/include/common/view-tests.hpp:22]: check_equal and unary_check are similar, I don't see a difference.

Comment (lslusarczyk) [reply test/gtest/include/common/view-tests.hpp:22]: `equal` is standard and has obvious semantic, reader has to see `unary_check` for implementation to see that 1st param is just for printing it out and 2nd, 3rd are for equality check

Comment (rscohn2) [reply test/gtest/include/common/view-tests.hpp:22]: I don't object if you want to port or adapt the range v3 tests or write tests in that style. I started writing the DR tests that way and found they weren't what I needed to find and fix the bugs. There are complex interactions between views and algorithms, off by 1 and other boundary conditions and I need to write tests that are flexible and combine views and algorithms to find the issues.
